### ISSUE TITLE
[BUG] KeyError: 'custom' at sf.model.ModelDict when using a custom model

### ISSUE THREAD

**Issue by lfbsdev** [jamesdolezal/slideflow#196]
## Description

When parsing any loaded model to `model` in `ModelParams`, a key error is raised:
```
model_fn = self.ModelDict[self.model]
KeyError: 'custom'
```

## To Reproduce
1.
 `import keras.applications as kapps`
2.
 ```
hp = ModelParams(
    tile_px=299,
    tile_um=299,
    model=kapps.Xception,
    batch_size=32,
    epochs=[1],
    normalizer='reinhard',
    augment=True,
)
```
3.
```
P.train(
    'category',
    params=hp,
)
```

```
Process SpawnProcess-7:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/opt/conda/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/conda/lib/python3.7/site-packages/slideflow/project_utils.py", line 114, in _train_worker
    results = trainer.train(train_dts, val_dts, **training_kw)
  File "/opt/conda/lib/python3.7/site-packages/slideflow/model/tensorflow.py", line 1580, in train
    checkpoint=checkpoint
  File "/opt/conda/lib/python3.7/site-packages/slideflow/model/tensorflow.py", line 541, in build_model
    num_classes, **kwargs, activation='softmax'
  File "/opt/conda/lib/python3.7/site-packages/slideflow/model/tensorflow.py", line 337, in _build_categorical_or_linear_model
    tile_image_model, model_inputs = self._build_base(pretrain)
  File "/opt/conda/lib/python3.7/site-packages/slideflow/model/tensorflow.py", line 285, in _build_base
    base_model = self._get_core(weights=pretrain)
  File "/opt/conda/lib/python3.7/site-packages/slideflow/model/tensorflow.py", line 234, in _get_core
    model_fn = self.ModelDict[self.model]
KeyError: 'custom'

```
## Expected behavior

Training with the provided model.

## Environment:

 - Slideflow Version (e.g., 1.0): 1.3.2
 - OS (e.g., Ubuntu): Ubuntu
 - How you installed Slideflow (`pip`, source): pip
 - Python version: 3.7
 - CUDA/cuDNN version: 8.1.0.77-1+cuda11.2
 - GPU models and configuration: P100
 - Any other relevant information: Kaggle Environment from 2022-09-15

**Comment by lfbsdev** [comment]
I took a look into the `ModelParams` class at `model.tensorflow.py` and it seems it inherits a `ModelDict` with `{'custom' : *model*}  ` from `model.base.py` and redefines it there. Maybe that's the problem?

**Comment by jamesdolezal** [comment]
Thanks for raising this issue! 

The root cause comes from pickling the `ModelParams` object as it is passed to the isolated process for training. Tensorflow (and PyTorch) modules cannot be pickled, so the custom module is lost. This bug was probably introduced after we made some changes to the internal structure of the `ModelParams` object.

A complete solution will require a more in-depth investigation into how this class is pickled, along with stability testing. In the meantime, I've pushed a fix that drops the process isolation when training with a custom model. If you run from the master branch, it should be fixed. I'll incorporate a final fix into the next version (1.3.3)

Thanks again for the report - let me know if you encounter any other issues.
